samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.cpp

```diff
@@ -0,0 +1,1015 @@
+	uniform_data.view = glm::scale(camera.matrices.view, glm::vec3(0.1f, -0.1f, 0.1f));
+	uniform_buffer->convert_and_update(uniform_data);
+}
+
+void DynamicMultisampleRasterization::draw()
+{
+	ApiVulkanSample::prepare_frame();
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+}
+
+void DynamicMultisampleRasterization::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+	draw();
+
+	if (camera.updated)
+	{
+		update_uniform_buffers();
+	}
+}
+
+void DynamicMultisampleRasterization::update_resources()
+{
+	prepared = false;
+
+	if (device)
+	{
+		device->wait_idle();
+
+		// Destroy color attachment
+		vkDestroyImageView(get_device().get_handle(), depth_stencil.view, nullptr);
+		vkDestroyImage(get_device().get_handle(), depth_stencil.image, nullptr);
+		vkFreeMemory(get_device().get_handle(), depth_stencil.mem, nullptr);
+		depth_stencil.view  = VK_NULL_HANDLE;
+		depth_stencil.image = VK_NULL_HANDLE;
+		depth_stencil.mem   = VK_NULL_HANDLE;
+
+		// Destroy depth/stencil attachment
+		vkDestroyImageView(get_device().get_handle(), color_attachment.view, nullptr);
+		vkDestroyImage(get_device().get_handle(), color_attachment.image, nullptr);
+		vkFreeMemory(get_device().get_handle(), color_attachment.mem, nullptr);
+		color_attachment.view  = VK_NULL_HANDLE;
+		color_attachment.image = VK_NULL_HANDLE;
+		color_attachment.mem   = VK_NULL_HANDLE;
+
+		if (render_pass != VK_NULL_HANDLE)
+		{
+			vkDestroyRenderPass(get_device().get_handle(), render_pass, nullptr);
+			render_pass = VK_NULL_HANDLE;
+		}
+
+		setup_render_pass();
+		setup_framebuffer();
+		rebuild_command_buffers();
+	}
+
+	prepared = true;
+}
+
+void DynamicMultisampleRasterization::on_update_ui_overlay(vkb::Drawer &drawer)
+{
+	if (drawer.header("Settings"))
+	{
+		if (drawer.combo_box("antialiasing", &gui_settings.gui_sample_count, gui_settings.sample_counts))
+		{
+			sample_count = supported_sample_count_list[gui_settings.gui_sample_count];
+
+			update_resources();
+		}
+	}
+}
+
+bool DynamicMultisampleRasterization::resize(const uint32_t _width, const uint32_t _height)
+{
+	if (!ApiVulkanSample::resize(_width, _height))
+		return false;
+
+	prepared = false;
+
+	update_resources();
+
+	prepared = true;
+	return true;
+}
+
+void DynamicMultisampleRasterization::input_event(const vkb::InputEvent &input_event)
+{
+	ApiVulkanSample::input_event(input_event);
+
+	bool gui_captures_event = false;
+
+	if (gui)
+	{
+		gui_captures_event = gui->input_event(input_event);
+	}
+
+	if (!gui_captures_event)
+	{
+		if (input_event.get_source() == vkb::EventSource::Keyboard)
+		{
+			const auto &key_button = static_cast<const vkb::KeyInputEvent &>(input_event);
+
+			if (key_button.get_action() == vkb::KeyAction::Down)
+			{
+				switch (key_button.get_code())
+				{
+					case vkb::KeyCode::Q:
```

Comment:
You should mention the support of key 'q' in the readme and/or some startup message.
Any why is it 'q' to switch through the msaa-modes?

Reply:
Sorry, this was commited by mistake, during my experiments. I will remove the whole chunk.